aquadoggo/src/network/service.rs

```diff
@@ -259,68 +419,48 @@ pub async fn network_service(
-                            trace!("NAT status changed from {:?} to {:?}", old, new);
+            }
 
-                            if let Some(addr) = external_circuit_addr.clone() {
-                                trace!("Adding external relayed listen address: {}", addr);
-                                swarm.add_external_address(addr, AddressScore::Finite(1));
+            // ~~~~~~
+            // Limits
+            // ~~~~~~
+            SwarmEvent::Behaviour(BehaviourEvent::Limits(event)) => {
+                debug!("Unhandled connection limit event: {event:?}")
+            }
 
-                                if let Some(rendezvous_peer_id) = network_config.rendezvous_peer_id
-                                {
-                                    // Invoke registration of relayed client address with the rendezvous server
-                                    if let Some(rendezvous_client) =
-                                        swarm.behaviour_mut().rendezvous_client.as_mut()
-                                    {
-                                        rendezvous_client.register(
-                                            rendezvous::Namespace::from_static(NODE_NAMESPACE),
-                                            rendezvous_peer_id,
-                                            None,
-                                        );
-                                    }
-                                }
-                            }
-                        }
-                        autonat::Event::InboundProbe(_) | autonat::Event::OutboundProbe(_) => (),
-                    }
+            // ~~~~~~~~~~~
+            // Replication
+            // ~~~~~~~~~~~
+            SwarmEvent::Behaviour(BehaviourEvent::Replication(event)) => match event {
+                replication::Event::MessageReceived(peer_id, message, connection_id) => self
+                    .send_service_message(ServiceMessage::ReceivedReplicationMessage(
+                        peer_id,
+                        connection_id,
+                        message,
+                    )),
+                replication::Event::ConnectionEstablished(peer_id, connection_id) => {
+                    // Inform other services about new connection
+                    self.send_service_message(ServiceMessage::ConnectionEstablished(
+                        peer_id,
+                        connection_id,
+                    ));
                 }
-                SwarmEvent::Behaviour(BehaviourEvent::Limits(event)) => {
-                    debug!("Unhandled connection limit event: {event:?}")
+                replication::Event::ConnectionClosed(peer_id, connection_id) => {
+                    // Inform other services about closed connection
+                    self.send_service_message(ServiceMessage::ConnectionClosed(
+                        peer_id,
+                        connection_id,
+                    ));
                 }
```

Comment:
This is where we catch the events triggered in the replication behaviour and pass them onto the service bus.